tests/components/rainforest_raven/test_diagnostics.py

```diff
@@ -0,0 +1,103 @@
+"""Test the Rainforest Eagle diagnostics."""
+from dataclasses import asdict
+
+import pytest
+
+from homeassistant.components.diagnostics import REDACTED
+from homeassistant.const import CONF_MAC
+from homeassistant.core import HomeAssistant
+
+from . import create_mock_device, create_mock_entry
+from .const import DEMAND, NETWORK_INFO, PRICE_CLUSTER, SUMMATION
+
+from tests.common import patch
+from tests.components.diagnostics import get_diagnostics_for_config_entry
+
+
+@pytest.fixture
+def mock_device():
+    """Mock a functioning RAVEn device."""
+    mock_device = create_mock_device()
+    with patch(
+        "homeassistant.components.rainforest_raven.coordinator.RAVEnSerialDevice",
```

Comment:
This looks like a copy from the coordinator tests. Common fixtures should go in a `conftest.py` module.